### PR TITLE
fix: 🐛 use pupBrowser.isConnected to check health

### PR DESCRIPTION
**Pull request by hcfw007**

### CODE REVIEW

**Review by bung87 — APPROVED** [review]

File: src/manager.ts [hunk 1]
```diff
@@ -577,7 +574,9 @@ export class Manager extends EventEmitter {
       case WAState.CONNECTED:
         this.clearPendingLogoutEmitTimer()
         this.emit('login', this.botId)
+        logger.info('will sync contact')
```
Comment (su-chang) [inline src/manager.ts:577]: I think this log is useless.

File: src/manager.ts [hunk 2]
```diff
@@ -577,7 +574,9 @@ export class Manager extends EventEmitter {
...
         const contactOrRoomList = await this.syncContactOrRoomList()
+        logger.info('contact synced')
```
Comment (su-chang) [inline src/manager.ts:579]: ditto

File: src/manager.ts [hunk 3]
```diff
@@ -993,4 +992,34 @@ export class Manager extends EventEmitter {
...
+
+  private stopHeartbeat () {
+    if (this.heartbeatTimer) {
+      clearInterval(this.heartbeatTimer)
+    }
+  }
+
+  private asystoleCount = 0
+  private async heartbeat () {
+    const alive = this.whatsAppClient?.pupBrowser?.isConnected()
```
Comment (su-chang) [inline src/manager.ts:1012]: It's better to use `this.getWhatsApp()` to get `this.whatsAppClient`.

File: src/manager.ts [hunk 4]
```diff
@@ -993,4 +992,34 @@ export class Manager extends EventEmitter {
...
+    if (alive) {
+      this.asystoleCount = 0
+    } else {
+      this.asystoleCount += 1
+      if (this.asystoleCount > 5) {
```
Comment (su-chang) [inline src/manager.ts:1017]: We should add an `CONST_VARIABLE` and `ENV` for number `5`.

**Review by su-chang — CHANGES_REQUESTED** [review]

File: src/manager.ts (hunk 1 above)
Comment (hcfw007) [reply src/manager.ts:577]: Indeed. I added this last night trying to find the process before ready event and forgot to remove it.

**Review by su-chang — APPROVED** [review]
LGTM

**Review by su-chang — APPROVED** [review]

**Pull request merged**